core/drivers/bcm_sotp.c

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#include <assert.h>
+#include <drivers/bcm_sotp.h>
+#include <initcall.h>
+#include <io.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+
+#define SOTP_PROG_CONTROL			0x0
+#define SOTP_PROG_CONTROL__OTP_CPU_MODE_EN	(1 << 15)
+#define SOTP_PROG_CONTROL__OTP_DISABLE_ECC	(1 << 9)
+#define SOTP_ADDR__OTP_ROW_ADDR_R		6
+
+#define SOTP_ADDR				0xc
+
+#define SOTP_CTRL_0				0x10
+#define SOTP_CTRL_0__START			(1 << 0)
+#define SOTP_READ				(0 << 1)
+
+#define SOTP_STAT_0				0x18
+#define SOTP_STATUS_0__FDONE			(1 << 3)
+
+#define SOTP_STATUS_1				0x1c
+#define SOTP_STATUS_1__CMD_DONE			(1 << 1)
+#define SOTP_STATUS_1__ECC_DET			(1 << 17)
+
+#define SOTP_RDDATA_0				0x20
+#define SOTP_RDDATA_1				0x24
+#define SOTP_ADDR_MASK				0x3ff
+
+#define SOTP_ECC_ERR_DETECT			0x8000000000000000
+
+#define MAX_TIME_OUT				30000
+
+struct bcm_sotp_dev *bcm_sotp;
+
+struct bcm_sotp_dev {
+	vaddr_t *sotp_base;
+};
+
+static int otp_status_done_wait(uint32_t addr, uint32_t bit)
+{
+	uint32_t timeout;
+	uint32_t reg_val;
+
+	timeout = 0;
+	do {
+		reg_val = io_read32(addr);
+		timeout++;
+	} while (((reg_val & bit) == 0) && timeout < MAX_TIME_OUT);
+
+	if (timeout == MAX_TIME_OUT)
+		return TEE_ERROR_BUSY;
+	return 0;
```

Comment:
Replace `0` with `TEE_SUCCESS` and change function prototype to return a `TEE_Result` value instead of an `int`.

Same comment for `bcm_iproc_sotp_mem_read()` below.